hugofs/glob.go

```diff
@@ -31,9 +31,24 @@ func Glob(fs afero.Fs, pattern string, handle func(fi FileMetaInfo) (bool, error
 		return nil
 	}
 
-	g, err := glob.GetGlob(pattern)
-	if err != nil {
-		return nil
+	hasUpperCase := pattern != strings.ToLower(pattern)
```

Comment:
So, this looks logically correct for this particular case, but one of my motivations behind my previous comment about this was to consolidate the "filename matching" to make sure that it behaved the same in the different places.

Which means that `glob.GetFilenamesGlob` should return an interface that does the double checking (see the struct that is put in filenamesGlobCache).

Reply:
Added new interface.
Is this ok?